pkg/extensions/search/utils/db.go

```diff
@@ -0,0 +1,244 @@
+package cveinfo
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"go.etcd.io/bbolt"
+)
+
+const (
+	// NvdDB ...
+	NvdDB = "NvdJSON"
+	// VendorDB ...
+	VendorDB = "NvdPkgVendor"
+	// NameDB ...
+	NameDB = "NvdPkgName"
+	// NameverDB ...
+	NameverDB = "NvdPkgNameVer"
+	// NvdmetaDB ...
+	NvdmetaDB = "NvdMeta"
+)
+
+/*Connect ... Create a database connection to */
+func (cve CveInfo) Connect(dbPath string) *bbolt.DB {
+	// Opening the connection on DB on given port
+	db, err := bbolt.Open(dbPath, 0600, nil)
+	if err != nil {
+		cve.Log.Error().Err(err).Msg("Not able to open a database")
+		return nil
+	}
+
+	return db
+}
+
+// nolint (interfacer)
+func Close(db *bbolt.DB) error {
+	err := db.Close()
+	return err
+}
+
+//CreateDB ...
+func (cve CveInfo) CreateDB(dbname string, db *bbolt.DB) bool {
```

Comment:
Where is this being used in real code above?

Reply:
Used in util.go InitSearch method